### PR TITLE
remove ops.relation refs

### PR DESCRIPTION
**Pull request by sed-i**

### CODE REVIEW

File: lib/charms/alertmanager_k8s/v0/alertmanager.py [hunk 1]
```diff
@@ -43,7 +42,25 @@ def restore(self, snapshot):
...
 
 
-class AlertmanagerConsumer(ConsumerBase):
+class AlertmanagerConsumerEvents(ObjectEvents):
+    """Event descriptor for events raised by `AlertmanagerConsumer`."""
+
+    cluster_changed = EventSource(ClusterChanged)
+
+
+class RelationManagerBase(Object):
```
Comment (balbirthomas) [inline lib/charms/alertmanager_k8s/v0/alertmanager.py:51]: It is not clear to me why you need this class.

File: lib/charms/karma_k8s/v0/karma.py [hunk 2]
```diff
@@ -165,22 +174,20 @@ class KarmaConsumer(ConsumerBase):
...
-    def __init__(self, charm, name: str, consumes: dict, multi: bool = False):
-        super().__init__(charm, name, consumes, multi)
+    def __init__(self, charm, name: str):
+        super().__init__(charm, name)
         self.charm = charm
 
         events = self.charm.on[self.name]
-        self.framework.observe(events.relation_changed, self._on_relation_changed)
-        self.framework.observe(events.relation_departed, self._on_relation_departed)
+        self.charm.framework.observe(events.relation_changed, self._on_relation_changed)
```
Comment (balbirthomas) [inline lib/charms/karma_k8s/v0/karma.py:189]: `Object` also has a `framework` instance. I am not entirely sure if this change is necessary.

**Review by balbirthomas — APPROVED** [review]

File: lib/charms/karma_k8s/v0/karma.py (hunk 2 above)
Comment (sed-i) [reply lib/charms/karma_k8s/v0/karma.py:189]: Leftover. Fixed.

File: lib/charms/alertmanager_k8s/v0/alertmanager.py (hunk 1 above)
Comment (sed-i) [reply lib/charms/alertmanager_k8s/v0/alertmanager.py:51]: Same is repeated for both provider and consumer. Reduce duplication and (re)establish contract.

**Pull request merged**